impl_core/src/test/java/io/opencensus/implcore/metrics/MetricRegistryImplTest.java

```diff
@@ -53,248 +54,63 @@
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addDoubleGauge(
-        null,
-        DESCRIPTION,
-        UNIT,
-        labels,
-        null,
-        new ToDoubleFunction<Object>() {
-          @Override
-          public double applyAsDouble(Object value) {
-            return 5.0;
-          }
-        });
-  }
+  // helper class
+  public static class TotalMemory {
+    public double getUsed() {
+      return 2.13;
+    }
 
-  @Test
-  public void addDoubleGauge_NullDescription() {
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addDoubleGauge(
-        NAME,
-        null,
-        UNIT,
-        labels,
-        null,
-        new ToDoubleFunction<Object>() {
-          @Override
-          public double applyAsDouble(Object value) {
-            return 5.0;
-          }
-        });
-  }
-
-  @Test
-  public void addDoubleGauge_NullUnit() {
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addDoubleGauge(
-        NAME,
-        DESCRIPTION,
-        null,
-        labels,
-        null,
-        new ToDoubleFunction<Object>() {
-          @Override
-          public double applyAsDouble(Object value) {
-            return 5.0;
-          }
-        });
-  }
-
-  @Test
-  public void addDoubleGauge_NullLabels() {
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addDoubleGauge(
-        NAME,
-        DESCRIPTION,
-        UNIT,
-        null,
-        null,
-        new ToDoubleFunction<Object>() {
-          @Override
-          public double applyAsDouble(Object value) {
-            return 5.0;
-          }
-        });
-  }
-
-  @Test
-  public void addDoubleGauge_NullFunction() {
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addDoubleGauge(NAME, DESCRIPTION, UNIT, labels, null, null);
-  }
-
-  @Test
-  public void addDoubleGauge_GetMetrics() {
-    metricRegistry.addDoubleGauge(
-        NAME,
-        DESCRIPTION,
-        UNIT,
-        labels,
-        null,
-        new ToDoubleFunction<Object>() {
-          @Override
-          public double applyAsDouble(Object value) {
-            return 5.0;
-          }
-        });
-    assertThat(metricRegistry.getMetricProducer().getMetrics())
-        .containsExactly(
-            Metric.create(
-                MetricDescriptor.create(
-                    NAME,
-                    DESCRIPTION,
-                    UNIT,
-                    Type.GAUGE_DOUBLE,
-                    Collections.unmodifiableList(Collections.singletonList(LABEL_KEY))),
-                Collections.singletonList(
-                    TimeSeries.createWithOnePoint(
-                        Collections.unmodifiableList(Collections.singletonList(LABEL_VALUES)),
-                        Point.create(Value.doubleValue(5.0), TEST_TIME),
-                        null))));
+    public long getAvailable() {
+      return 7;
+    }
   }
 
   @Test
   public void addLongGauge_NullName() {
```

Comment:
Add a test to make sure that you get a LongGaugeImpl. So check that the return object is instance of this class.

Reply:
Done